src/products/services/products.service.ts

```diff
@@ -14,11 +16,13 @@ export class ProductsService {
     private readonly productRepository: ProductsRepository,
   ) {}
 
-  async getAll(): Promise<ProductEntity[]> {
+  async getAll(
+    query: PaginationQueryDTO,
+  ): Promise<Pagination<ProductEntity[]>> {
     try {
       this.logger.log(`Attempting to find all products`);
-      const products = await this.productRepository.findAll();
-      this.logger.log(`Founded ${products.length} products`);
+      const products = await this.productRepository.findAll(query);
```

Comment:
Can you validate if that query has a correct format? And also create the test to check the behavior

Reply:
What do you mind with "has a correct format"?

Reply:
Im taking about check if `query` has correct parameters and and has no unwanted parameters

Reply:
Well about correct parameters, I'm using the class validator annotations to check that. About unwanted parameters, I think the actual code will ignore the queries that I'm not using. Is that ok?

Reply:
Sure!